src/main/java/com/coveo/spillway/trigger/AbstractLimitTrigger.java

```diff
@@ -56,13 +60,26 @@ public AbstractLimitTrigger(LimitTriggerCallback callback) {
    * @return True if the limit is triggered, false otherwise
    */
   protected abstract <T> boolean triggered(
-      T context, int cost, int currentLimitValue, LimitDefinition limitDefinition);
+      T context, int currentLimitValue, LimitDefinition limitDefinition);
 
   @Override
   public <T> void callbackIfRequired(
-      T context, int cost, int currentLimitValue, LimitDefinition limitDefinition) {
-    if (triggered(context, cost, currentLimitValue, limitDefinition)) {
+      T context,
+      int cost,
+      Instant timestamp,
+      int currentLimitValue,
+      LimitDefinition limitDefinition) {
+    Instant currentBucket = calculateBucket(timestamp, limitDefinition.getExpiration());
+
+    if (triggered(context, currentLimitValue, limitDefinition)
```

Comment:
Shouldn't we allow the counter to be incremented beyond the limit? That would solve this issue.